Indicators/BetaIndicator.cs

```diff
@@ -0,0 +1,171 @@
+        /// </summary>
+        public int WarmUpPeriod { get; private set; }
+
+        /// <summary>
+        /// Gets a flag indicating when the indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => _symbolDataPoints.Values.All(x => x.IsReady);
+
+        /// <summary>
+        /// Creates a new BetaIndicator with the specified name, period, target and 
+        /// reference values
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="period">The period of this indicator</param>
+        /// <param name="targetSymbol">The target symbol of this indicator</param>
+        /// <param name="referenceSymbol">The reference symbol of this indicator</param>
+        public BetaIndicator(string name, int period, Symbol targetSymbol, Symbol referenceSymbol)
+            : base(name)
+        {
+            // Assert the period is greater than two, otherwise the beta can not be computed
+            if (period < 2)
+            {
+                throw new Exception($"Period parameter for Beta indicator must be greater than 2 but was {period}");
+            }
+
+            WarmUpPeriod = period;
+            _referenceSymbol = referenceSymbol;
+            _targetSymbol = targetSymbol;
+
+            _symbolDataPoints = new Dictionary<Symbol, RollingWindow<decimal>>();
+            _symbolDataPoints.Add(_referenceSymbol, new RollingWindow<decimal>(WarmUpPeriod));
+            _symbolDataPoints.Add(_targetSymbol, new RollingWindow<decimal>(WarmUpPeriod));
+
+            _targetReturns = new RollingWindow<double>(WarmUpPeriod);
+            _referenceReturns = new RollingWindow<double>(WarmUpPeriod);
+            _beta = 0;
+        }
+
+        /// <summary>
+        /// Computes the next value for this indicator from the given state.
+        /// 
+        /// As this indicator is receiving data points from two different symbols,
+        /// it's going to compute the next value when the amount of data points
+        /// of each of them is the same. Otherwise, it will return the last beta
+        /// value computed
+        /// </summary>
+        /// <param name="input">The input value of this indicator on this time step.
+        /// It can be either from the target or the reference symbol</param>
+        /// <returns>The beta value of the target used in relation with the reference</returns>
+        protected override decimal ComputeNextValue(TradeBar input)
+        {
+            var inputSymbol = input.Symbol;
+            _symbolDataPoints[inputSymbol].Add(input.Close);
+
+            if (_symbolDataPoints[_targetSymbol].Samples == _symbolDataPoints[_referenceSymbol].Samples && _symbolDataPoints[_targetSymbol].Samples > 1)
+            {
+                _targetReturns.Add(GetNewReturn(_symbolDataPoints[_targetSymbol]));
+                _referenceReturns.Add(GetNewReturn(_symbolDataPoints[_referenceSymbol]));
```

Comment:
Suggest storing `_symbolDataPoints[_referenceSymbol]` & `_symbolDataPoints[_targetSymbol]` in a local variable 👍